modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/resource/ResourceManagerBean.java

```diff
@@ -783,6 +789,82 @@ private boolean uninventoryResourceBulkDeleteAsyncWork(Subject overlord, int res
                 resourceIds);
         }
 
+        // If this resource had packages installed, remove their version and bits
+        if (installedPackages.size() > 0) {
+            hasErrors |= cleanOrphanedPackages(overlord, installedPackages);
```

Comment:
If we are not going to remove the Package (if it has no more PackageVersions) then this should probably be renamed cleanOrphanedPackageVersions().